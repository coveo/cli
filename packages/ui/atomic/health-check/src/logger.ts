import chalk from 'chalk';

export function log(...messages: string[]) {
  for (const message of messages) {
    console.log(message);
  }
}

export function groupStart() {
  console.group();
}

export function groupEnd() {
  console.groupEnd();
}

export function newLine() {
  console.log();
}

export function success(message: string) {
  log([chalk.green('✔'), message].join(' '));
}

export function failure(message: string) {
  log([chalk.red('✖'), message].join(' '));
}
