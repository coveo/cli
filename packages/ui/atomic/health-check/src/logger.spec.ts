import {stdout} from 'stdout-stderr';
import {failure, log, newLine, success} from './logger';

describe('Logger', () => {
  beforeEach(() => {
    stdout.start();
  });

  afterEach(() => {
    stdout.stop();
  });

  it('#log should print multiple lines to stdout', () => {
    log('Maecenas purus lacus', 'Donec ut sem cursus');
    expect(stdout.output).toMatchSnapshot();
  });

  it('#success should print a successful message', () => {
    success('You win!');
    expect(stdout.output).toMatchSnapshot();
  });

  it('#failure should print an error message', () => {
    failure('Better luck next time');
    expect(stdout.output).toMatchSnapshot();
  });

  it('#newLine should print a new line', () => {
    newLine();
    expect(stdout.output).toMatchSnapshot();
  });
});
