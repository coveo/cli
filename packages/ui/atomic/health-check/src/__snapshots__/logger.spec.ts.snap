// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logger #failure should print an error message 1`] = `
"✖ Better luck next time
"
`;

exports[`Logger #log should print multiple lines to stdout 1`] = `
"Maecenas purus lacus
Donec ut sem cursus
"
`;

exports[`Logger #newLine should print a new line 1`] = `
"
"
`;

exports[`Logger #success should print a successful message 1`] = `
"✔ You win!
"
`;
